src/pint/toa.py

```diff
@@ -231,9 +231,15 @@ def get_TOAs(
             updatepickle = True
         else:
             if hasattr(t, "hashes"):
-                if not t.check_hashes():
+                try:
+                    if not t.check_hashes():
+                        updatepickle = True
+                        log.warning("Pickle file is based on files that have changed")
+                except FileNotFoundError:
```

Comment:
This check is a good idea - we don't want crashes! maybe we should catch exceptions more aggressively, like if the file exists but isn't readable - but perhaps we should make the paths absolute before storing the hashes, so that their stored location doesn't depend on the current directory? `os.path.abspath()` or one of the methods of Path objects should do that. A test would be nice too - make a pickle in a temporary directory and then try to change directory and access it.

Reply:
Maybe just accept that pickles have to be regenerated if you move around for now - but you shouldn't error out! - and file a bug describing the problem?